### DISCUSSION TITLE
Has anyone added a probe to a laser module?

### DISCUSSION THREAD

**Discussion by ReggiePye** [arkypita/LaserGRBL#2434] [General]
I want to add a probe to a laser module to help with focusing.
I have ordered a capacitive sensor to give it a go.
I think that on the Uno board I use A5 for the probe and if I can determine the material distance accurately from the laser module I should be able to set an offset for the Z position. I could then use some macros to reposition for various thicknesses of material that I would be cutting etc.
This is all in theory, so I wondered if anyone has some experience with this setup.

**Comment by StuartB4** [comment]
Hi Reggie. I personally use a USB MICROSCOPE  for focussing. The upside is
I don't have to look at the laser dot to focus, just watch it on my PC
screen. The downside is I have to do it everytime for varying thicknesses
of material, so can't use any macros.  I also use an UNO with a Protoneer
CNC shield for the laser controller. I'm sure someone on the Arduino
website would be able to help with your idea.
I've had loads of help there in the past.

**Reply by ReggiePye** [reply]
Thanks for that Stuart, I also have some cameras I could use as well, I have one on my CNC mill that was an inspection camera, very small and a long cable. These cameras are not expensive.
Industrial Endoscope Camera Ip67 Waterproof 5.5mm 7mm 3 In1 USB Borescope Inspection Snake Camera For Android Phone PC

I really want to be able to use the macros to speed up the setup processes, so I will concentrate on the probe idea, but could also fit a camera as well.

**Reply by Hostaler** [reply]
It's a very good idea and it would be great to be able to do it. :)

**Reply by ReggiePye** [reply]
I will share my experiences here when I get the sensor that I have ordered.
